### PR TITLE
PR raised for JBTM-3122 and JBTM-3132

### PR DESCRIPTION
**Pull request by mayankkunwar**
https://issues.redhat.com/browse/JBTM-3132
https://issues.redhat.com/browse/JBTM-3122

This PR changes includes a common profiles (one of such profile is Arquillian JDK9+ which defines JVM arguments needed for smooth run of tests, see https://issues.jboss.org/projects/JBTM/issues/JBTM-3122)

Refactoring of quickstarts to have the common parent

### CODE REVIEW

File: ArjunaJTA/jee_transactional_app/pom.xml [hunk 1]
```diff
@@ -13,6 +13,13 @@
...
+
+    <parent>
+        <groupId>org.jboss.narayana.quickstart.jta</groupId>
+        <artifactId>narayana-quickstarts-jta</artifactId>
+        <version>5.10.5.Final-SNAPSHOT</version>
+    </parent>
+
     <groupId>org.jboss.narayana.quickstart.jta</groupId>
     <artifactId>jee_transactional_app</artifactId>
     <version>5.10.5.Final-SNAPSHOT</version>
```
Comment (ochaloup) [inline ArjunaJTA/jee_transactional_app/pom.xml]: please, do not use the `<version>` twice in the `pom.xml`. Here you define the artifact `<artifactId>jee_transactional_app` which inherits the version from the parent.

It means the version for the artifact needs to be removed from all(*!*) the `pom.xml` where you added the `<parent>`.

**Comment by ochaloup** [comment]
@mayankkunwar more on this PR. Could you please elaborate why we don't need the `parent` at

- `./ArjunaJTS/interop/pom.xml`
- `./ArjunaJTS/standalone/pom.xml`
- `../ArjunaJTS/trailmap/pom.xml`
- `./ArjunaJTS/jts/pom.xml`
- `./rts/lra/pom.xml`
- `./rts/lra-examples/pom.xml`
- `./rts/at/...`
- `./STM/*/pom.xml`
- `./karaf/osgi-jta/pom.xml`
- `./transactionaldriver/transactionaldriver-and-tomcat/pom.xml`
- `./ArjunaJTA/maven/pom.xml`

Now the next "step" is that there would be good to unify the `<dependencyManagement>`. In the most cases there is imported Java EE spec. I think there should be used in parent Java EE 7 like. 
```
<dependencyManagement>
    <dependencies>
        <dependency>
            <groupId>org.jboss.spec</groupId>
            <artifactId>jboss-javaee-7.0</artifactId>
            <version>${version.javaee.spec}</version>
            <type>pom</type>
            <scope>import</scope>
        </dependency>
    </dependencies>
</dependencyManagement>
```
But there is needed to check if this will provide all necessary versions of artifacts.

File: ArjunaJTA/jee_transactional_app/pom.xml (hunk 1 above)
Comment (mayankkunwar) [reply ArjunaJTA/jee_transactional_app/pom.xml]: okay, I will remove it from all the POM files

**Comment by mayankkunwar** [comment]
@ochaloup I will add the dependencies as you mentioned in the above comments.
Some of the modules in which I have not added parent is because, I thought we were just using it to add the new property for arquillian tests for JDK9+, and in those modules there were no arquillian tests present, but as per your above comments, we need to add a common dependency javaee-7.0 at parent POM so that it could provide required artifacts, therefore now I will add parent to all the remaining modules.

**Comment by ochaloup** [comment]
@mayankkunwar I see. I should mention that point as well. The usefulnes of common parent is not only in definition of JDK profile but it could be use to unify e.g. the Arquillian profile (some day in future) and to unify the dependencies. I think going with the Java EE dependency for all the quickstarts is still doable as part of this PR. But if you think to leave it aside as a next task we can do so.
It's just that I haven't realize that I should mention this idea before. I can see you've already done some changes in this regard. But if you want feel free to return the PR back to the original state and we can move that task to the next iteration.
From that I understand now as well why some of the modules were avoided from the common parent. That makes sense.

File: ArjunaJTS/interop/pom.xml [hunk 2]
```diff
@@ -51,9 +52,44 @@
...
+      <artifactId>jboss-javaee-8.0</artifactId>
+      <version>1.0.4.Final</version>
+      <type>pom</type>
+    </dependency>
+  </dependencies>
+  <profiles>
+    <profile>
+      <id>include-for-JDK11-above-versions</id>
+      <activation>
+        <jdk>[11,)q</jdk>
```
Comment (jbliznak) [inline ArjunaJTS/interop/pom.xml:74]: looks like typo `q`

Comment (mayankkunwar) [reply ArjunaJTS/interop/pom.xml:74]: Thanks

**Comment by mayankkunwar** [comment]
Closing this due to rebase issue.

**Pull request closed without merging**